src/shogun/machine/IterativeLinearMachine.h

```diff
@@ -0,0 +1,68 @@
+/*
+ * This software is distributed under BSD 3-clause license (see LICENSE file).
+ *
+ * Authors: Shubham Shukla
+ */
+
+#ifndef _ITERATIVELINEARMACHINE_H__
+#define _ITERATIVELINEARMACHINE_H__
+
+#include <shogun/lib/config.h>
+
+#include <shogun/lib/SGVector.h>
+#include <shogun/lib/common.h>
+#include <shogun/machine/LinearMachine.h>
+
+namespace shogun
+{
+	class CFeatures;
+	class CLabels;
+
+	class CIterativeLinearMachine : public CLinearMachine
+	{
+	public:
+		/** Default constructor */
+		CIterativeLinearMachine();
+		virtual ~CIterativeLinearMachine();
+
+		/** Train classifier
+		  *
+		  * @param data training data (parameter can be avoided if distance or
+		  * kernel-based classifiers are used and distance/kernels are
+		  * initialized with train data)
+		  *
+		  * @return whether training was successful
+		  */
+		virtual bool train_machine(CFeatures* data = NULL);
```

Comment:
train_machine is protected btw. Be careful when inheriting to respect the base class structure!